src/NovelAudioService.cpp

```diff
@@ -0,0 +1,367 @@
+  }
+  return _channelMap[soundName];
+}
+
+void NovelAudioService::setSoundVolume(std::string soundName, float value) {
+  Mix_VolumeChunk(_sounds[soundName], convertToMixVolume(value));
+}
+
+void NovelAudioService::setSoundPosition(std::string soundName, int angle, int distance) {
+  if (_channelMap.find(soundName) != _channelMap.end() && _channelMap[soundName] != -1)
+  {
+    Mix_SetPosition(_channelMap[soundName], angle, distance);
+  }
+  else
+  {
+    std::cerr << soundName << " is not allocated to a channel already!" << std::endl;
+  }
+}
+
+void NovelAudioService::setSoundDistance(std::string soundName, int distance) {
+    if (_channelMap.find(soundName) != _channelMap.end() && _channelMap[soundName] != -1)
+  {
+    Mix_SetDistance(_channelMap[soundName], distance);
+  }
+  else
+  {
+    std::cerr << soundName << " is not allocated to a channel already!" << std::endl;
+  }
+}
+
+void NovelAudioService::setSoundPanning(std::string soundName, int leftChannelVolume, int rightChannelVolume) {
+    if (_channelMap.find(soundName) != _channelMap.end() && _channelMap[soundName] != -1)
+  {
+    Mix_SetPanning(_channelMap[soundName], leftChannelVolume, rightChannelVolume);
+  }
+  else
+  {
+    std::cerr << soundName << " is not allocated to a channel already!" << std::endl;
+  }
+}
+
+void NovelAudioService::loadMusic(std::string input) {
+  auto exists = _music.find(input);
+  if (exists != _music.end()) return;
+
+  Mix_Music* newMusic = Mix_LoadMUS(input.c_str());
+  if (newMusic != nullptr)
+  {
+    _music[input] = newMusic;
+  }
+  else
+  {
+    std::cerr << Mix_GetError() << std::endl;
+  }
+}
+
+void NovelAudioService::unloadMusic(std::string input) {
+  auto existingMusic = _music.find(input);
+  if (existingMusic == _music.end()) return;
+
+  Mix_FreeMusic(_music[input]);
+  _music.erase(existingMusic);
+}
+
+void NovelAudioService::playMusic() {
+  if (Mix_PausedMusic()) Mix_ResumeMusic();
+}
+
+void NovelAudioService::playMusic(std::string musicName) {
+  auto existingMusic = _music.find(musicName);
+  if (existingMusic == _music.end())
+  {
+    loadMusic(musicName);
+    existingMusic = _music.find(musicName);
+    if (existingMusic == _music.end()) return;
+  }
+  _channelMap[musicName] = -3;
+  Mix_PlayMusic(_music[musicName],-1);
+  _musicTime = SDL_GetPerformanceCounter();
+}
+
+void NovelAudioService::playMusic(std::string musicName, int loops) {
+
+  auto existingMusic = _music.find(musicName);
+  if (existingMusic == _music.end())
+  {
+    loadMusic(musicName);
+    existingMusic = _music.find(musicName);
+    if (existingMusic == _music.end()) return;
+  }
+  _channelMap[musicName] = -3;
+  Mix_PlayMusic(_music[musicName], loops-1);
+}
+
+void NovelAudioService::pauseMusic() {
+  if (!Mix_PausedMusic())
+  {
+    Mix_PauseMusic();
+    _musicPausedTime = SDL_GetPerformanceCounter();
+  }
+}
+
+void NovelAudioService::stopMusic() {
+  Mix_HaltMusic();
+  _musicPausedTime = SDL_GetPerformanceCounter();
```

Comment:
`Stop` and `Pause` seem like two different things...

Reply:
Let me remove that....

Reply:
Removed 👍